app/Exceptions/Auth/TokenGenerationException.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Exceptions\Auth;
+
+use Exception;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+
+final class TokenGenerationException extends Exception
+{
+    public function render(Request $request): JsonResponse
+    {
+        return new JsonResponse(
+            [
+                'error' => 'There was an error registering the user with the provided details',
+                'status_code' => Response::HTTP_BAD_REQUEST,
```

Comment:
Is there a specific requirement to include the status code in the body if its already set in the response as the actual status code?

Reply:
removed